File: main/src/main/java/com/crossapplication/main/repository/interfaces/WaterLogRepository.java
package com.crossapplication.main.repository.interfaces;

import java.time.LocalDate;
import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import com.crossapplication.main.entity.WaterLog;

@Repository
public interface WaterLogRepository extends JpaRepository<WaterLog, Long> {
    List<WaterLog> findByUserIdAndLogDate(Long userId, LocalDate logDate);

    @Query("SELECT COALESCE(SUM(w.amountMl),0) FROM WaterLog w WHERE w.user.id = :userId AND w.logDate = :logDate")
    double sumAmountByUserIdAndLogDate(@Param("userId") Long userId, @Param("logDate") LocalDate logDate);

    List<WaterLog> findByUserIdAndLogDateBetween(Long userId, LocalDate start, LocalDate end);
}

File: flutter/flutterfrontend/lib/domain/repositories/water_log_repository.dart
abstract class WaterLogRepository {
  Future<List<Map<String, dynamic>>> findByUserIdAndLogDate(
    int userId,
    DateTime logDate,
  );
  Future<double> sumAmountByUserIdAndLogDate(int userId, DateTime logDate);
  Future<List<Map<String, dynamic>>> findByUserIdAndLogDateBetween(
    int userId,
    DateTime start,
    DateTime end,
  );
  Future<Map<String, dynamic>?> findById(int id);
  Future<void> save(Map<String, dynamic> waterLog);
  Future<void> update(int id, Map<String, dynamic> waterLog);
  Future<void> delete(int id);
}
